Consider marking UnitTestFrameworkNamer.Name as virtual

My suggestion is to make this property virtual. This way it can be overridden.

https://github.com/approvals/ApprovalTests.Net/blob/0c2a590adc317be47fa452a063169f18d18fd08f/ApprovalTests/Namers/UnitTestFrameworkNamer.cs#L29

The reason for this is that I have a shared project with multiple target frameworks (e.g. NET45, NET46, NET47). Then I can postfix this and we can test per-framework as well without having to re-implement the whole base class.

btw here was the original build error:

```
9>C:\CI_WS\Ws\108434\Source\Orc_Squirrel\src\Orc.Squirrel.Tests\Orc.Squirrel.Tests.NET45\App_Packages\ApiApprover.6.0.0\PublicApiApprover.cs(30,36,30,40): error CS0506: 'PublicApiApprover.AssemblyPathNamer.Name': cannot override inherited member 'UnitTestFrameworkNamer.Name' because it is not marked virtual, abstract, or override
```

--- a/ApprovalTests.MSTest/ApprovalTests.MSTest.csproj
+++ b/ApprovalTests.MSTest/ApprovalTests.MSTest.csproj
@@ -11,4 +11,7 @@
     <ProjectReference Include="..\ApprovalUtilities\ApprovalUtilities.csproj" />
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
 </Project>

--- a/ApprovalTests.NHibernate/ApprovalTests.NHibernate.csproj
+++ b/ApprovalTests.NHibernate/ApprovalTests.NHibernate.csproj
@@ -16,9 +16,9 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="NHibernate" Version="4.1.1.4000" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssests="All" ExcludeAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="System" Pack="False" />
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />
   </ItemGroup>

--- a/ApprovalTests.NHibernate/ApprovalTests.NHibernate.csproj
+++ b/ApprovalTests.NHibernate/ApprovalTests.NHibernate.csproj
@@ -15,9 +15,9 @@
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Fody" Version="2.4.6"  PrivateAssets="All" />
     <PackageReference Include="NHibernate" Version="4.1.1.4000" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.5.0"  PrivateAssets="All" />
     <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="System" Pack="False" />
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />

--- a/ApprovalTests.Rdlc/ApprovalTests.Rdlc.csproj
+++ b/ApprovalTests.Rdlc/ApprovalTests.Rdlc.csproj
@@ -17,9 +17,9 @@
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssests="All" ExcludeAssets="All" />
-     <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssests="All" ExcludeAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="Microsoft.ReportViewer.WinForms, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" Pack="False">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\lib\ReportViewer\Microsoft.ReportViewer.WinForms.dll</HintPath>

--- a/ApprovalTests.Rdlc/ApprovalTests.Rdlc.csproj
+++ b/ApprovalTests.Rdlc/ApprovalTests.Rdlc.csproj
@@ -17,8 +17,8 @@
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.6" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.5.0" PrivateAssets="All" />
     <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="Microsoft.ReportViewer.WinForms, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" Pack="False">
       <SpecificVersion>False</SpecificVersion>

--- a/ApprovalTests.WinForms/ApprovalTests.WinForms.csproj
+++ b/ApprovalTests.WinForms/ApprovalTests.WinForms.csproj
@@ -15,8 +15,8 @@ WinFormsApprovals.Verify(new Button { BackColor = Color.LightBlue, Text = "Help"
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.6" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.5.0" PrivateAssets="All" />
     <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="System.Windows.Forms" />
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />

--- a/ApprovalTests.WinForms/ApprovalTests.WinForms.csproj
+++ b/ApprovalTests.WinForms/ApprovalTests.WinForms.csproj
@@ -15,9 +15,9 @@ WinFormsApprovals.Verify(new Button { BackColor = Color.LightBlue, Text = "Help"
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssests="All" ExcludeAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="System.Windows.Forms" />
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />
     <ProjectReference Include="..\ApprovalUtilities\ApprovalUtilities.csproj" />

--- a/ApprovalTests.Xunit/ApprovalTests.Xunit.csproj
+++ b/ApprovalTests.Xunit/ApprovalTests.Xunit.csproj
@@ -12,4 +12,7 @@
     <ProjectReference Include="..\ApprovalTests\ApprovalTests.csproj" />
     <ProjectReference Include="..\ApprovalUtilities\ApprovalUtilities.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
 </Project>

--- a/ApprovalTests/ApprovalTests.csproj
+++ b/ApprovalTests/ApprovalTests.csproj
@@ -14,9 +14,9 @@
   <ItemGroup>
     <PackageReference Include="AlphaFS" Version="2.1.3" />
     <PackageReference Include="EntityFramework" Version="5.0.0" PrivateAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.6" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.5.0" PrivateAssets="All" />
     <PackageReference Include="StatePrinter" Version="1.0.3" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
     <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <PackageReference Include="Microsoft.AspNet.Mvc" Version="3.0.20105.1" ExcludeAssets="All" />
     <PackageReference Include="Microsoft.AspNet.Razor" Version="1.0.20105.408" ExcludeAssets="All" />

--- a/ApprovalTests/ApprovalTests.csproj
+++ b/ApprovalTests/ApprovalTests.csproj
@@ -15,13 +15,13 @@
     <PackageReference Include="AlphaFS" Version="2.1.3" />
     <PackageReference Include="EntityFramework" Version="5.0.0" PrivateAssets="All" />
     <PackageReference Include="StatePrinter" Version="1.0.3" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssests="All" ExcludeAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <PackageReference Include="Microsoft.AspNet.Mvc" Version="3.0.20105.1" ExcludeAssets="All" />
     <PackageReference Include="Microsoft.AspNet.Razor" Version="1.0.20105.408" ExcludeAssets="All" />
     <PackageReference Include="Microsoft.AspNet.WebPages" Version="1.0.20105.408" ExcludeAssets="All" />
     <PackageReference Include="Microsoft.Web.Infrastructure" Version="1.0.0.0" ExcludeAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssests="All" ExcludeAssets="All" />
     <Reference Include="PresentationCore" Pack="False">
       <RequiredTargetFramework>3.0</RequiredTargetFramework>
     </Reference>

--- a/ApprovalUtilities/ApprovalUtilities.csproj
+++ b/ApprovalUtilities/ApprovalUtilities.csproj
@@ -13,10 +13,10 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="AlphaFS" Version="2.1.3" />
+    <PackageReference Include="Fody" Version="2.4.6" PrivateAssets="All" />
     <PackageReference Include="NHibernate" Version="4.*" PrivateAssets="All" />
     <PackageReference Include="Mvc2" Version="2.*" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.5.0" PrivateAssets="All" />
     <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="PresentationCore" Pack="False" />
     <Reference Include="PresentationFramework" Pack="False" />

--- a/ApprovalUtilities/ApprovalUtilities.csproj
+++ b/ApprovalUtilities/ApprovalUtilities.csproj
@@ -15,9 +15,9 @@
     <PackageReference Include="AlphaFS" Version="2.1.3" />
     <PackageReference Include="NHibernate" Version="4.*" PrivateAssets="All" />
     <PackageReference Include="Mvc2" Version="2.*" PrivateAssets="All" />
-    <PackageReference Include="Fody" Version="2.4.2" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssests="All" ExcludeAssets="All" />
-    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssests="All" ExcludeAssets="All" />
+    <PackageReference Include="Fody" Version="2.4.2" PrivateAssets="All" />
+    <PackageReference Include="Publicize.Fody" Version="1.4.0" PrivateAssets="All" />
+    <PackageReference Include="Virtuosity.Fody" Version="1.20.2" PrivateAssets="All" />
     <Reference Include="PresentationCore" Pack="False" />
     <Reference Include="PresentationFramework" Pack="False" />
     <Reference Include="System.Core" Pack="False" />

